silo/tests/test_readview.py

```diff
@@ -0,0 +1,157 @@
+        view = ReadViewSet.as_view({'get': 'list'})
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 4)
+
+    def test_list_read_normal_user(self):
+        request = self.factory.get('/api/read/')
+        request.user = self.tola_user.user
+        view = ReadViewSet.as_view({'get': 'list'})
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 0)
+
+        factories.Read(owner=self.tola_user.user)
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+
+
+class ReadRetrieveViewTest(TestCase):
+    def setUp(self):
+        factories.Organization(id=1)
+        self.user = factories.User(first_name='Homer', last_name='Simpson')
+        self.tola_user = factories.TolaUser(user=self.user)
+        self.read = factories.Read(read_name="test_data",
+                                   owner=self.tola_user.user)
+        self.silo = factories.Silo(owner=self.tola_user.user,
+                                   reads=[self.read])
+        self.factory = APIRequestFactory()
+
+    def test_retrieve_read_superuser(self):
+        request = self.factory.get('/api/read/')
+        request.user = factories.User.build(is_superuser=True,
+                                            is_staff=True)
+        view = ReadViewSet.as_view({'get': 'retrieve'})
+        response = view(request, pk=self.read.id)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['read_name'], self.read.read_name)
+
+    def test_retrieve_read_normal_user(self):
+        request = self.factory.get('/api/read/')
+        request.user = factories.User.build(is_superuser=False,
+                                            is_staff=False)
+        view = ReadViewSet.as_view({'get': 'retrieve'})
+        response = view(request, pk=self.read.id)
+        self.assertEqual(response.status_code, 404)
+
+        request = self.factory.get('/api/read/')
+        read = factories.Read(read_name='My Read', owner=self.tola_user.user)
+        request.user = self.tola_user.user
+
+        response = view(request, pk=read.id)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['read_name'], read.read_name)
+
+
+class ReadDataViewTest(TestCase):
+    def setUp(self):
+        self.factory = APIRequestFactory()
+        self.tola_user = factories.TolaUser()
+
+        self.read = factories.Read(read_name="test_data",
+                                   owner=self.tola_user.user)
+
+        self.read_type = factories.ReadType.create(read_type='CSV')
+
+    def test_create_read(self):
+        data = {
+            "type": "/api/readtype/"+str(self.read_type.id),
+            "owner": "/api/users/"+str(self.tola_user.user.id),
```

Comment:
Use the `reverse` instead. It's better to don't hardcode the URL.
```python
reverse('readytype-detail',
      kwargs={'pk': self.read_type.id},
      request=request)
```

Reply:
It didn't work at first, so i hardcoded it, i will check the error again.